### ISSUE TITLE
Handle migration for `addon-docs/blocks` better

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#20149]
We removed the `@storybook/addon-docs/blocks` entry in 7.0. There's a MIGRATION.md entry, but no automigration for MDX files.

This is bad because it currently fails with the following inscrutable error in `dev` mode:

```
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "[object Object]".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

We should:
- [x] Handle the error better https://github.com/storybookjs/storybook/issues/20161
- [x] Potentially add back `addon-docs/blocks` as deprecated
- [ ] Potentially create an migration to move to `blocks` library instead

**Comment by Semigradsky** [comment]
I don't use `@storybook/addon-docs/blocks` but I have the same error (like `UnhandledPromiseRejection`) and it is really not clear what's wrong. So `Handle the errors better` would be nice 👍

**Comment by shilman** [comment]
@Semigradsky filed the error handling issue separately here with workaround https://github.com/storybookjs/storybook/issues/20161#issuecomment-1343097966

**Comment by shilman** [comment]
Whoopee!! I just released https://github.com/storybookjs/storybook/releases/tag/v7.0.0-beta.7 containing PR #20246 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.

**Comment by shilman** [comment]
@yannbf can you please take the automigration here?

**Comment by yannbf** [comment]
For the automigration, here are possible steps:

1. Change export from `/index` of addon-docs to re-export from`./blocks` so we get the deprecation message

2. Automigrate `.stories.mdx` (only) files:
```ts
import { Canvas } from '@storybook/addon-docs'
import { Canvas } from '@storybook/addon-docs/blocks'
```

Which should become
```ts
import { Canvas } from '@storybook/blocks'
```

3. Add `@storybook/blocks` dependency if blocks are used

**Comment by kasperpeulen** [comment]
Fixed by #21073
